kube-core/src/params.rs

```diff
@@ -438,6 +441,46 @@ impl PostParams {
 /// };
 /// let patch = Patch::Apply(&r);
 /// ```
+/// # Invalid Patches
+///
+/// In the example below the patch contains a `PodSpec` and **not** a complete or partial `Pod`.
+/// It is therefore an invalid patch as the full structure of a resource, `Pod` in this case, is required when
+/// patching. The invalid patch will be accepted by the K8S API, but no changes will be made to the cluster.
+///
+/// [`PatchParams::validation_strict`] can cover for this case.
+///
+/// An example with an invalid and valid patch:
+///
+/// ```no_run
+/// use k8s_openapi::api::core::v1::{Pod, PodSpec};
+/// use kube::{Api, api::{PatchParams, Patch}};
+///
+/// # async fn wrapper() -> Result<(), Box<dyn std::error::Error>> {
+/// # let client = kube::Client::try_default().await?;
+/// let pods: Api<Pod> = Api::namespaced(client, "apps");
+/// let pp = PatchParams::default();
+///
+/// let invalid_patch: PodSpec = serde_json::from_value(serde_json::json!({
+///                 "activeDeadlineSeconds": 5
+/// }))?;
+///
+/// // This will have no effect on mypod.
+/// pods.patch("mypod", &pp, &Patch::Strategic(invalid_patch)).await?;
+///
+/// let valid_patch: Pod = serde_json::from_value(serde_json::json!({
+///           "spec": {
+///                 "activeDeadlineSeconds": 5
+///            }
+/// }))?;
```

Comment:
This is non-serverside style patch that will not work without `metadata.name` and typemeta (aka a complete `Pod` struct equivalent).

The recommendation here is to use `validation_strict` only makes sense to me when we are doing serverside apply. If this is a failure of `Strategic` or `Merge` then it's strange that it works (as in bubbles up an error?) when we use `:;validation_strict()` :thinking:

Reply:
Ah yes,  I see what you mean :facepalm: I will go over this again.

Reply:
After testing again non-server side patches do appear to be affected by the strict validation. The error returned does not appear to originate from field validation but rather that the cluster can not process the incorrect(which is correct). Maybe this is some side effect of enabling strict validation. I can not find anything indidicating this behavior in the [docs](https://kubernetes.io/docs/reference/using-api/api-concepts/#validation-for-unrecognized-or-duplicate-fields-setting-the-field-validation-level).

I don't think investigating this behavior is not relevant for this PR, so I will dump my findings into a separate issue.

```rust
        let invalid_patch: PodSpec = serde_json::from_value(serde_json::json!({
                        "activeDeadlineSeconds": 5
        }))?;

        pods.patch(
            &pod_name,
            &PatchParams::default().validation_strict(),
            &Patch::Strategic(invalid_patch),
        )
        .await?;

```

I get the following error:

```
Error: Api(ErrorResponse { status: "Failure", message: " \"\" is invalid: patch: Invalid value: \"map[activeDeadlineSeconds:5 containers:[]]\": strict decoding error: unknown field \"activeDeadlineSeconds\", unknown field \"containers\"", reason: "Invalid", code: 422 })
```

Reply:
I think the best thing for the example in the docs is to just use serve-side patches. That also is inline with the general recommendation presented in the docs as well.

Reply:
I have applied these changes and attempted to clean up the formulation a bit. Let me know what you think :)

Reply:
> After testing again non-server side patches do appear to be affected by the strict validation. The error returned does not appear to originate from field validation. The 422 indicates that the cluster can not process the incorrect object(which is accurate). Maybe this is some side effect of enabling strict validation. I can not find anything indicating this behavior in the [docs](https://kubernetes.io/docs/reference/using-api/api-concepts/#validation-for-unrecognized-or-duplicate-fields-setting-the-field-validation-level).
> 
> I don't think investigating this behavior is relevant for this PR, I will dump my findings into a separate issue for later. Feel free to object.
> 
> ```rust
>         let invalid_patch: PodSpec = serde_json::from_value(serde_json::json!({
>                         "activeDeadlineSeconds": 5
>         }))?;
> 
>         pods.patch(
>             &pod_name,
>             &PatchParams::default().validation_strict(),
>             &Patch::Strategic(invalid_patch),
>         )
>         .await?;
> ```
> 
> I get the following error:
> 
> ```
> Error: Api(ErrorResponse { status: "Failure", message: " \"\" is invalid: patch: Invalid value: \"map[activeDeadlineSeconds:5 containers:[]]\": strict decoding error: unknown field \"activeDeadlineSeconds\", unknown field \"containers\"", reason: "Invalid", code: 422 })
> ```
 
 
Follow up, performing the same test with server-side patch returns a 400 complaining about patch version. This is more inline with the docs I think. As the strict validation is supposed to return a 400.

```
Api(ErrorResponse { status: "Failure", message: "Incorrect version specified in apply patch. Specified patch version: , expected: v1", reason: "BadRequest", code: 400 })
```

Reply:
> I think the best thing for the example in the docs is to just use serve-side patches. That also is inline with the general recommendation presented in the docs as well.

Sorry for dragging this out :sweat: 

Using server-side patching mitigates the issue of incorrect patches, as the patch is rejected due to a missing patch version. Therefore I have reverted the example to use a strategic patch and noted that serve-side patches avoid this issue.

> This is non-serverside style patch that will not work without `metadata.name` and typemeta (aka a complete `Pod` struct equivalent).

This does not appear to be case :thinking: unless kube-rs automatically adds `metadata.name` I can perform non server-side patches just fine without.

Reply:
> > This is non-serverside style patch that will not work without `metadata.name` and typemeta (aka a complete `Pod` struct equivalent).
> 
> This does not appear to be case thinking unless kube-rs automatically adds `metadata.name`. I can perform non server-side patches just fine without.

 Kube does not add data to patches anywhere. Maybe I am misremembering this entirely. Looking over this now I see `pod_api` example doing a `PatchParams::default` with a `Patch::Merge` without a `metadata.name`.

I am unable to verify this right now due to a desktop issue, but at any rate, we do not need to block the PR over this. You have sorted out everything for the subresource, the `Api` impl, and everything seems good on CI. 

How do you feel about picking out the `params.rs` docs changes into its own PR? That is the only thing that I feel warrants one extra verification. Am happy to merge the remaining changes.

Reply:
That sounds perfect. `params.rs` needs a second pair of eyes.

Reply:
Done: https://github.com/kube-rs/kube/pull/1262